frontend/src/Frontend/Crypto/Class.hs

```diff
@@ -15,19 +15,44 @@ import Obelisk.Route.Frontend
 import Reflex.Dom hiding (fromJSString)
 import Reflex.Host.Class (MonadReflexCreateTrigger)
 import Data.ByteString (ByteString)
+import Data.Text (Text)
+
+import Pact.Types.Scheme (PPKScheme)
 
 import Frontend.Crypto.Ed25519
 import Frontend.Foundation
 import Frontend.Storage
 
 instance (HasStorage m, Monad m) => HasStorage (CryptoT key m)
+-- TODO : Hide the pact key constructor so the caller is forced to verify it
+data PactKey = PactKey
+  { _pactKey_scheme :: PPKScheme
+  , _pactKey_publicKey :: PublicKey
+  , _pactKey_secret :: ByteString
+  } deriving Show
+
+data GenKeyArg
+  = GenWalletIndex Int
+  | GenFromPactKey PactKey
 
 data Crypto key = Crypto
   { _crypto_sign :: ByteString -> key -> JSM Signature
-  , _crypto_genKey :: Int -> JSM (key, PublicKey)
+  , _crypto_genKey :: GenKeyArg -> JSM (key, PublicKey)
+  , _crypto_verifyPactKey :: PPKScheme -> Text -> JSM (Either String PactKey)
```

Comment:
This seems like it doesn't belong here. Does this function change between desktop/web?

Reply:
Yes, it does change: on desktop we use the function from Pact.Types.Crypto which is only available on GHC.